crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -59,30 +67,52 @@ enum AuctionIdRecoveryStatus {
 }
 
 impl OnSettlementEventUpdater {
-    pub async fn run_forever(self) -> ! {
-        let mut current_block = self.eth.current_block().borrow().to_owned();
-        let mut block_stream = ethrpc::current_block::into_stream(self.eth.current_block().clone());
+    pub fn new(eth: infra::Ethereum, db: Postgres) -> Self {
```

Comment:
Doc comment mentioning that this immediately kicks off the indexing process would be nice.